drivers/SmartThings/matter-switch/src/init.lua

```diff
@@ -250,17 +257,14 @@ end
 local function find_default_endpoint(device, component)
   local switch_eps = device:get_endpoints(clusters.OnOff.ID)
   local button_eps = device:get_endpoints(clusters.Switch.ID, {feature_bitmap=clusters.Switch.types.Feature.MOMENTARY_SWITCH})
-  local all_eps = {}
 
-  for _,ep in ipairs(switch_eps) do
-    table.insert(all_eps, ep)
-  end
+  -- use first button endpoint as main endpoint if one exists
   for _,ep in ipairs(button_eps) do
-    table.insert(all_eps, ep)
+    if ep ~= 0 then --0 is the matter RootNode endpoint
```

Comment:
I think you would need to sort the button endpoints first. This was done in like 261 in the previous code after all eps were combined into one table FYI. In this case, we may need to sort the button and switch eps separately and then return the relevant ep.

Reply:
Good call - I made this change in the latest commit.